Source/OCMock/OCMockObject.m

```diff
@@ -261,50 +284,75 @@ - (void)forwardInvocation:(NSInvocation *)anInvocation
     }
     @catch(NSException *e)
     {
-        [exceptions addObject:e];
+        @synchronized(exceptions)
+        {
+            [exceptions addObject:e];
+        }
         [e raise];
     }
 }
 
 - (BOOL)handleInvocation:(NSInvocation *)anInvocation
 {
-    [invocations addObject:anInvocation];
-
+    @synchronized(invocations)
+    {
+        // When the method has a char* argument we do not retain the arguments. This makes it possible
+        // to match char* args literally and with anyPointer. Not retaining the argument means that
+        // in these cases tests that use their own autorelease pools may fail unexpectedly.
+        if(![anInvocation hasCharPointerArgument])
+            [anInvocation retainArguments];
+        [invocations addObject:anInvocation];
+    }
+    
```

Comment:
Yes...  Yuck.  The target is very likely to be self, and it's used too so there's no clearing it.  Beargh.  The logical place to clear it would be in -stopMocking, but people don't currently call that method typically.  It would seem the options are these.

1. Release note it and change the documentation to say that -stopMocking always has to be called manually, and the object won't deallocate until you do, a la NSPort and NSTimer.
2. Have people set something in their unit test bundle's Info.plist that says they want some thread safety with OCMock.
3. Use a global or something like that and make people manually enable thread safe behavior on OCMock as a whole.

Personally I prefer the first option; I don't really want to manually enable thread safety, it should just be there.  I also kind of feel like 2 and 3 are jumping through a lot of hoops just to save on a few -stopMocking calls.